include/roboteam_ai/stp/Tactic.h

```diff
@@ -30,24 +28,11 @@ class Tactic {
      *
      * Though this method is responsible for ensuring everything is calculated, it helps to use helpers so this
      * function doesn't become a massive hack
+     * @param info StpInfo struct that contains the info passed down from Role
+     * @return an StpInfo struct with extra information for the skill
      */
     virtual std::optional<StpInfo> calculateInfoForSkill(StpInfo const &info) noexcept = 0;
 
-    /**
-     * called on initialization of this tactic
-     */
-    virtual void onInitialize() noexcept = 0;
-
-    /**
-     * called on update of this tactic
-     */
-    virtual void onUpdate(Status const &status) noexcept = 0;
-
-    /**
-     * called on terminate of this tactic
-     */
-    virtual void onTerminate() noexcept = 0;
-
```

Comment:
I assume you removed this because the onupdates do not have a purpose, right?

Reply:
Yes, to make the structure more clear. And besides, STP should not be stateful and these functions kind of lead the developer to believe they can be.